src/flb_sds.c

```diff
@@ -306,6 +310,30 @@ flb_sds_t flb_sds_cat_utf8 (flb_sds_t *sds, const char *str, int str_len)
         head = FLB_SDS_HEADER(s);
     }
 
+    while (1) {
+        offset = head->len;
+        ret = flb_utils_write_str(s, &offset, flb_sds_alloc(s), str, str_len);
+        if (ret == FLB_FALSE) {
+            /* realloc */
+            size = flb_sds_alloc(s) * 2;
+            tmp = flb_sds_increase(s, size);
+            if (tmp == NULL) {
+                return NULL;
+            }
+            *sds = s = tmp;
+            head = FLB_SDS_HEADER(s);
+        }
+        else {
+            break;
+        }
+    }
+
+    flb_sds_len_set(s, offset);
+    s[head->len] = '\0';
+    return s;
+
+
+
     for (i = 0; i < str_len; i++) {
```

Comment:
Can we remove the latter part of flb_sds_cat_utf8?
Or just leaving as-is for now to be able to restore easily?

Reply:
thanks, I will do the cleanup shortly